code/game/objects/effects/decals/Cleanable/misc_cleanables.dm

```diff
@@ -141,12 +141,77 @@
 	gender = PLURAL
 	density = FALSE
 	layer = TURF_LAYER
+	plane = FLOOR_PLANE
 	icon = 'icons/effects/blood.dmi'
 	icon_state = "vomit_1"
 	random_icon_states = list("vomit_1", "vomit_2", "vomit_3", "vomit_4")
 	no_clear = TRUE
 	scoop_reagents = list("vomit" = 5)
 
+
+/obj/effect/decal/cleanable/vomit/Initialize(mapload)
+	. = ..()
+	var/turf/T = get_turf(loc)
+	gravity_check = has_gravity(src, T)
+	if(get_turf(src) != loc)
```

Comment:
any reason in particular you get the turf of the loc instead of `src`? I ask because if there's not really a good reason you should just replace `get_turf(src)` with `T` and do `var/turf/T = get_turf(src)` instead